src/apps/chifra/internal/scrape/scrape_consolidate.go

```diff
@@ -109,7 +104,7 @@ func (bm *BlazeManager) Consolidate(blocks []base.Blknum) (error, bool) {
 			publisher := base.ZeroAddr
 			var chunk index.Chunk
 			if report, err := chunk.Write(chain, publisher, chunkPath, appMap, nAppearances); err != nil {
-				return err, false
```

Comment:
This is the one place where the forever loop in the caller would have quit. We tried to write the file, it failed, but the file may still exist and be partial. In the C++ code, we would have been writing to a temporary file and only after it completed would be move it to the actual chunk. We don't do that here. (That's probably a bug.)

I think there may be two things to do:

1. Can we cancel the context here so that the forever loop exits?
2. Can we write to a temporary file and only if the write suceeds, move the temporary file on to the chunk?

Let's discuss

Reply:
I think I should bring back the `ok`. That would be the first step and later we can add a temporary file. 
Or, and I think it'd be the most idiomatic way, we have a custom error type, like `CriticalScraperError`. If we see this kind of error, we have to abort. This pattern could be used elsewhere, too and it's easier to integrate than changing method definitions

Reply:
I like the idea of a custom error, but I question whether we're actually doing the right thing. For all other paths, we "clean up" (I think it's called `removeEphemeralFiles`). This means remove the stage. In this case, the error happened during creation of the chunk, so I think this is why it's different. In the previous code, we quit, but I don't think that's necessarily the right thing to do. We know the error happened during writing of the chunk, so unlike all other errors, we fear that the file may be incomplete.

Instead of quiting, we could perhaps just delete the chunk file (and allow the cleaning of ephemeral files to continue as normal).

Also, it's curious to me that we don't have a similar test during the writing of the bloom file.

Also, we should double check that the `backup file mechanism` isn't already in place.